src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java

```diff
@@ -229,14 +229,18 @@ String computeCrc32c(String input) {
                 break;
             }
 
+            final long retryWait = SLEEP_BASE_MILLISECONDS * Double.valueOf(Math.pow(2.5, noProgressCount)).longValue();
             if (completeCount == 0) {
                 noProgressCount++;
                 if (noProgressCount > firestoreRetries) {
                     throw new FileSystemExecutionException("batch operation failed. " +
                         firestoreRetries + " tries with no progress.");
+                } else {
+                    logger.info("will attempt retry #{} after {} millisecond pause.", noProgressCount, retryWait);
```

Comment:
Totally up to you - Would it be worth it to translate the units here to some easier to read? Like seconds or minutes?

Reply:
Yeah, was wondering.  Given that it's logging, I might just leave it as millis